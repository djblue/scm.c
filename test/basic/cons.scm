(write (pair? (cons 1 2)))
(write (car (cons 1 2)))
(write (cdr (cons 1 2)))

(write (pair? '(1 . 2)))
(write (car '(1 . 2)))
(write (cdr '(1 . 2)))

(define p (cons '() '()))

(write (car p))
(set-car! p 1)
(write (car p))

(write (cdr p))
(set-cdr! p 2)
(write (cdr p))
