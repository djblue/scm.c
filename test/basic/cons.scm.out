#t
1
2
#t
1
2
nil
1
nil
2
